# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/webkitgtk/default-uri.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.h.vanilla	2021-07-29 17:05:38.096130563 +0200
+++ webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.h	2021-07-29 17:05:57.046129807 +0200
@@ -37,7 +37,7 @@
 #define BROWSER_IS_WINDOW(obj)         (G_TYPE_CHECK_INSTANCE_TYPE((obj), BROWSER_TYPE_WINDOW))
 #define BROWSER_IS_WINDOW_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE((klass),  BROWSER_TYPE_WINDOW))
 #define BROWSER_WINDOW_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS((obj),  BROWSER_TYPE_WINDOW, BrowserWindowClass))
-#define BROWSER_DEFAULT_URL            "http://www.webkitgtk.org/"
+#define BROWSER_DEFAULT_URL            "http://t2sde.org/"
 #define BROWSER_ABOUT_SCHEME           "minibrowser-about"
 
 typedef struct _BrowserWindow        BrowserWindow;
--- webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.c.vanilla	2021-07-29 17:27:47.389077560 +0200
+++ webkitgtk-2.32.3/Tools/MiniBrowser/gtk/BrowserWindow.c	2021-07-29 17:29:07.242074376 +0200
@@ -94,13 +94,20 @@
 
 static void activateUriEntryCallback(BrowserWindow *window)
 {
-    browser_window_load_uri(window,
+    const char* uri =
 #if GTK_CHECK_VERSION(3, 98, 0)
-        gtk_editable_get_text(GTK_EDITABLE(window->uriEntry))
+        gtk_editable_get_text(GTK_EDITABLE(window->uriEntry));
 #else
-        gtk_entry_get_text(GTK_ENTRY(window->uriEntry))
+        gtk_entry_get_text(GTK_ENTRY(window->uriEntry));
 #endif
-    );
+    const char* proto = strstr(uri, "://");
+    if (!proto) {
+	char uri2[strlen("https://") + strlen(uri) + 1];
+	snprintf(uri2, sizeof(uri2), "https://%s", uri);
+	browser_window_load_uri(window, uri2);
+    } else {
+	browser_window_load_uri(window, uri);
+    }
 }
 
 static void reloadOrStopCallback(GSimpleAction *action, GVariant *parameter, gpointer userData)
